solidity/contracts/bridge/Bridge.sol

```diff
@@ -306,35 +356,445 @@ contract Bridge is Ownable {
+        // returns `0`, so we jump over one byte of compactSize uint.
         //
-        // TODO We need to validate `sweepTx.outputVector` to see if the balance
-        //      was not transferred away from the wallet before increasing
-        //      balances in the bank.
+        // For >= 253 && <= 0xffff there is `0xfd` tag,
+        // `BTCUtils.determineVarIntDataLengthAt` returns `2` (no
+        // tag byte included) so we need to jump over 1+2 bytes of
+        // compactSize uint.
         //
-        // TODO Delete deposit from unswept mapping or mark it as swept
-        //      depending on the gas costs. Alternatively, do not allow to
-        //      use the same TX input vector twice. Sweep should be provable
-        //      only one time.
+        // Please refer `BTCUtils` library and compactSize uint
+        // docs in `BitcoinTx` library for more details.
+        uint256 inputStartingIndex = 1 + inputsCompactSizeUintLength;
+
+        // Determine the swept deposits count. If main UTXO is NOT expected,
+        // all inputs should be deposits. If main UTXO is expected, one input
+        // should point to that main UTXO.
+        depositors = new address[](
+            !mainUtxoExpected ? inputsCount : inputsCount - 1
+        );
+        depositedAmounts = new uint256[](depositors.length);
+
+        // Initialize helper variables.
+        uint256 processedDepositsCount = 0;
+
+        // Inputs processing loop.
+        for (uint256 i = 0; i < inputsCount; i++) {
+            // Check if we are at the end of the input vector.
+            if (inputStartingIndex >= sweepTxInputVector.length) {
+                break;
+            }
+
+            (bytes32 inputTxHash, uint32 inputTxIndex, uint256 inputLength) =
+                parseTxInputAt(sweepTxInputVector, inputStartingIndex);
+
+            DepositInfo storage deposit =
+                deposits[
+                    uint256(
+                        keccak256(abi.encodePacked(inputTxHash, inputTxIndex))
+                    )
+                ];
+
+            if (deposit.revealedAt != 0) {
+                // If we entered here, that means the input was identified as
+                // a revealed deposit.
+                require(deposit.sweptAt == 0, "Deposit already swept");
+
+                if (processedDepositsCount == depositors.length) {
+                    // If this condition is true, that means a deposit input
+                    // took place of an expected main UTXO input.
+                    // In other words, there is no expected main UTXO
+                    // input and all inputs come from valid, revealed deposits.
+                    // In that case, we should ignore that input and let the
+                    // transaction fail at `require` checking whether expected
+                    // main UTXO was found.
+                    inputStartingIndex += inputLength;
+                    continue;
```

Comment:
Actually, why do we let the loop continue instead of reverting here?

Reply:
Good point! There was a difference at the beginning but since we introduced some other checks during development we can revert as you noticed. See: https://github.com/keep-network/tbtc-v2/pull/106/commits/dde3398b5d85298c42e99bd04e5e45041709e91f